pkg/leader/leaderelector.go

```diff
@@ -43,6 +46,10 @@ func (f *DummyElector) IsLeader() bool {
 	return true
 }
 
+func (f *DummyElector) RunWithLeader(ctx context.Context, run func() error) error {
+	return nil
```

Comment:
Why not
return run()
And than the logic is the same for on-prem and k8s

Reply:
+1

Reply:
cool, nice